### PR TITLE
Add base CLI

### PR DESCRIPTION
**Pull request by matheusfigueredo**
The CLI should provide an easy way for users to execute quick actions (write a note, add a goal, etc).

Examples of usages:
``` 
$ > ./target/debug/trad echo -m "Not so bad, not so bad"
Echo -> Not so bad, not so bad
$ >
$ > ./target/debug/trad -d echo -m "58 points? Yeah, absolutely!"
CliOpt { command: Echo { message: "58 points? Yeah, absolutely!" }, debug: true }
Echo -> This is a test message

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+extern crate core;
```
Comment (VanGrx) [inline src/lib.rs]: Do we need `lib` file?

File: src/bin/cli.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+struct CliOpt {
+    #[structopt(subcommand)]
+    command: Command,
+
+    // -d or --debug, true if it is present
+    #[structopt(short = "d", long = "debug")]
+    debug: bool,
+}
+
+fn main() {
```
Comment (VanGrx) [inline src/bin/cli.rs]: Can we add this part with args as part of main.rs?
In that case we can run binary without args in interactive mode and with args for quick note.

Comment (matheusfigueredo) [reply src/bin/cli.rs]: Done!

File: src/lib.rs (hunk 1 above)
Comment (matheusfigueredo) [reply src/lib.rs]: As the handlers are under `handlers` folder, if we remove it, it won't be able to find the module.

This line that you mention isn't necessary, I just removed in my last commit.

Comment (VanGrx) [reply src/lib.rs]: Yes, but adding
`mod handles;`
to main.rs will be enough as we will be building only bin.

**Review by VanGrx — APPROVED** [review]
LGTM!

**Pull request merged**